File: DemoProjectKMP-Public/End/KMP_footballmatches/shared/src/commonMain/kotlin/com/example/kmp_footballmatches/domain/LeagueRepository.kt
package com.example.kmp_footballmatches.domain

import com.example.kmp_footballmatches.data.model.League


/**
 * Created by Menes SIMEU on 10/06/2024.
 * Volvo MoCE Android Team member
 */
interface LeagueRepository {
    suspend fun getLeagues(): List<League>
}

File: DemoProjectKMP-Public/Start/iOS/iOS_footballmatches/iOS_footballmatches/domain/LeagueRepository.swift
//
//  LeagueRepository.swift
//  iOS_footballmatches
//
//  Created by Menes  SIMEU on 06/06/2024.
//

import Foundation

protocol LeagueRepository {
    func getLeagues(completion: @escaping (Result<[League], Error>) -> Void)
}
